### PR TITLE
[SCSE-231] Add types package

### PR DESCRIPTION
**Pull request by RealDyllon**

### CODE REVIEW

**Comment by RealDyllon** [comment]
@jamiegoh this change affects cms

@chanbakjsd should we require CODEOWNERS to approve PR if their code is affected by another change? (in this case jamie's cms code is changed)

**Review by chanbakjsd — APPROVED** [review]

**Pull request merged**
